lib/main.js

```diff
@@ -70,7 +70,23 @@ globals.setImmediate(() => {
 	runner.on('test', test);
 
 	process.on('ava-run', options => {
-		runner.run(options).then(exit);
+		runner.run(options)
+			.then(exit)
+			.catch(err => {
+				let exception;
+				try {
+					exception = serializeError(err);
+				} catch (_) {
+					// Avoid using serializeError
+					const err = new Error('Runner failed with an unserializable exception');
```

Comment:
I wonder, should we put this logic in `serializeError` so it handles reporting about unserializable exceptions everywhere?